templates/al2/runtime/max-pods-calculator.sh

```diff
@@ -116,7 +116,7 @@ if [[ "$CNI_MAJOR_VERSION" -gt 1 ]] || ([[ "$CNI_MAJOR_VERSION" = 1 ]] && [[ "$C
   PREFIX_DELEGATION_SUPPORTED=true
 fi
 
-DESCRIBE_INSTANCES_RESULT=$(aws ec2 describe-instance-types --instance-type "${INSTANCE_TYPE}" --query 'InstanceTypes[0].{Hypervisor: Hypervisor, EniCount: NetworkInfo.MaximumNetworkInterfaces, PodsPerEniCount: NetworkInfo.Ipv4AddressesPerInterface, CpuCount: VCpuInfo.DefaultVCpus}' --output json)
+DESCRIBE_INSTANCES_RESULT=$(aws ec2 describe-instance-types --instance-type "${INSTANCE_TYPE}" --query 'InstanceTypes[0].{Hypervisor: Hypervisor, EniCount: NetworkInfo.NetworkCards[0].MaximumNetworkInterfaces, PodsPerEniCount: NetworkInfo.Ipv4AddressesPerInterface, CpuCount: VCpuInfo.DefaultVCpus}' --output json)
```

Comment:
Do we always need network card 0, or the one specified by `NetworkInfo.DefaultNetworkCardIndex`?

I don't remember seeing an instance type that doesn't have `DefaultNetworkCardIndex = 0`, and I assume the `NetworkInfo.NetworkCards` will be sorted by their `NetworkCardInex`, but we may want to add handling for this anyway?

Reply:
> the one specified by NetworkInfo.DefaultNetworkCardIndex

Yeah, you are right. We are using the one specified by `DefaultNetworkCardIndex` here - https://github.com/aws/amazon-vpc-cni-k8s/blob/master/scripts/gen_vpc_ip_limits.go#L152

Colloquially, we say Network card at Index 0.  Agree to your suggestion on change.

Reply:
Makes sense. Updated to use DefaultNetworkCardIndex